indico/modules/events/registration/controllers/management/fields.py

```diff
@@ -158,6 +161,11 @@ def _process_DELETE(self):
             raise BadRequest
         signals.event.registration_form_field_deleted.send(self.field)
         self.field.is_deleted = True
+        db.session.add(self.field)
+        if 'show_if_field_id' in (self.field.data or {}):
+            self.field.data.pop('show_if_field_id')
+            self.field.data.pop('show_if_field_value')
+            flag_modified(self.field, 'data')
```

Comment:
I don't think this is necessary / correct. You're removing the condition from, the field which you are deleting. But instead, you would need to remove it from any field that references this one. TBH, better to just disallow the deletion like we already do for disabling. I'll add a commit for this, but unless I'm missing something here this code block should be removed.

Reply:
I was forced to add that line. Otherwise changes were not reflected in the database. That being said, if you have an alternative execution path, go ahead.

Reply:
The comment is on the whole block you added, it was mainly about the fact that with disallowing deletion this block of code is no longer needed since you no longer need to make changes to the data.

Reply:
Hmmmm do you consider OK to end up with deleted fields that keep in storage these show-if data? There is no actual database removal, but simply a boolean that says so.

Reply:
I think it's fine as long as we ignore them in `is_condition`. Makes it easier to recover from fuckups (someone accidentally deleting a field).

But after thinking about it again I'd also be OK with keeping this code here, since it removes one potential source of bugs (restoring a field and not realizing that it was used in some convoluted condition which was modified in the meantime).

The `db.session.add(self.field)` isn't necessary though. `flag_modified` is enough as anything already stored in the DB is also part of the DB session.